### ISSUE TITLE
Smarter defaults for network and training hyperparameters

### ISSUE THREAD

**Issue by theoheimel** [madgraph-ml/madnis#8]
Make the defaults for network and training hyperparameters smarter. Some ideas:

- adapt flow sub-network and CWnet dimensions based on integrand dimension
- choose amount of buffered training based on relative computational cost of network and integrand (see also #7)
- batch size, maybe depending on number of channels like in the MadNIS reloaded paper
- learning rate scheduling (ideally without having to know the training size in advance)
- Some other features like stratified training and channel dropping are off by default. Do we want to change that?
- other ideas?
